drivers/sensor/apds9960/apds9960.c

```diff
@@ -71,15 +74,24 @@ static int apds9960_sample_fetch(const struct device *dev,
 		LOG_ERR("Power on bit not set.");
 		return -EIO;
 	}
-
 	k_sem_take(&data->data_sem, K_FOREVER);
+#endif
 #endif
 
 	if (i2c_reg_read_byte_dt(&config->i2c,
 			      APDS9960_STATUS_REG, &tmp)) {
 		return -EIO;
 	}
 
+#ifdef CONFIG_APDS9960_FETCH_MODE_POLL
+	while (!(tmp & APDS9960_STATUS_PINT) && !(tmp & APDS9960_STATUS_AINT)) {
```

Comment:
Potential infinite loop? Can we add a way that this loop resolves in -ETIMEDOUT?